app/src/main/java/org/oppia/app/recyclerview/RecyclerDataDiffCallback.kt

```diff
@@ -0,0 +1,23 @@
+package org.oppia.app.recyclerview
+
+import androidx.recyclerview.widget.DiffUtil
+
+class RecyclerDataDiffCallback<T>(var oldList: List<T>, var newList: List<T>) :
+  DiffUtil.Callback() {
+
+  override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
```

Comment:
Could we check the item types here instead of contents? We don't have another way to quickly check equivalence, but as I understand it we want areItemsTheSame() to be fast, whereas areContentsTheSame can be a bit more expensive.

Reply:
Ok, `areContentTheSame` only called for those items whose `areItemsTheSame` gives true.

Reply:
**Could we check the item types here instead of contents?** This is amazing.

Reply:
Done